x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/sort/BucketedSortCommon.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data.sort;
+
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.BitArray;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.search.sort.SortOrder;
+
+/**
+ * Components common to BucketedSort implementations.
+ */
+class BucketedSortCommon implements Releasable {
```

Comment:
I yanked this out because it looked like it'd be safe to share at least a little code. I didn't plug this into the `X-BucketedSort` classes yet. But I think it's just about the same thing.